openwisp_controller/config/base/vpn.py

```diff
@@ -72,8 +84,12 @@ def save(self, *args, **kwargs):
         if not self.cert:
             self.cert = self._auto_create_cert()
         if not self.dh:
-            self.dh = self.dhparam(2048)
+            self.dh = self._placeholder_dh
+        is_adding = self._state.adding
         super().save(*args, **kwargs)
+        if is_adding:
+            with transaction.atomic():
+                transaction.on_commit(lambda: create_vpn_dh.delay(self.id))
```

Comment:
Theoretically a `Vpn` object may be get the DH passed as a parameter, which in this case would be overwritten.

Just do this above when you set `self.dh = self._placeholder_dh`.
The background task will be scheduled after the transaction is committed anyway.

```python
if not self.dh:
    self.dh = self._placeholder_dh
    with transaction.atomic():
        # ...
```